### PR TITLE
feat(linter): implement no-return-assign

### PR DESCRIPTION
**Pull request by radu2147**
This PR implements the not-recomennded eslint rule `no-return-assign`
https://github.com/oxc-project/oxc/issues/479

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/no_return_assign.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                return;
+            }
+            let mut parent_node = ctx.nodes().parent_node(node.id());
+            while parent_node.is_some_and(|parent| !is_sentinel_node(parent.kind())) {
+                parent_node = ctx.nodes().parent_node(parent_node.unwrap().id());
+            }
+            if parent_node.is_none() {
+                return;
+            }
+            let parent = parent_node.unwrap();
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_return_assign.rs:79]: Don't unwrap, just return false if it's None

File: crates/oxc_linter/src/rules/eslint/no_return_assign.rs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            match parent.kind() {
+                AstKind::ReturnStatement(_) => {
+                    ctx.diagnostic(no_return_assign_diagnostic(
+                        parent.span(),
+                        "Return statement should not contain assignment.",
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_return_assign.rs:84]:
```suggestion
                        "Return statement should not contain an assignment.",
```

File: crates/oxc_linter/src/rules/eslint/no_return_assign.rs [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                    ));
+                }
+                AstKind::ArrowFunctionExpression(_) => {
+                    ctx.diagnostic(no_return_assign_diagnostic(
+                        parent.span(),
+                        "Arrow function should not return assignment.",
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_return_assign.rs:90]:
```suggestion
                        "Arrow function should not return an assignment.",
```

File: crates/oxc_linter/src/rules/eslint/no_return_assign.rs [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+pub struct NoReturnAssign {
+    always_disallow_assignment_in_return: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallows assignment operators in return statements
+    ///
+    /// ### Why is this bad?
+    /// Assignment is allowed by js in return expression, but usually, an expression with only one equal sign is intended to be a comparison.
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_return_assign.rs:23]:
```suggestion
    /// Assignment is allowed by js in return expressions, but usually, an expression with only one equal sign is intended to be a comparison.
```

**Review by DonIsaac — COMMENTED** [review]
Thanks for the PR! Looking pretty good, just a few small changes and we can merge this.

File: crates/oxc_linter/src/rules/eslint/no_return_assign.rs [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// () => (a = b)
+    /// function x() { var result = a = b; return result; }
+    /// ```
+    NoReturnAssign,
+    style, // See <https://oxc.rs/docs/contribute/linter.html#rule-category> for details
+    suggestion  // Options are 'fix', 'fix_dangerous', 'suggestion', and 'conditional_fix_suggestion'
```
Comment (camc314) [inline crates/oxc_linter/src/rules/eslint/no_return_assign.rs:42]:
```suggestion
    style,
    suggestion

**Review by camc314 — APPROVED** [review]

**Review by DonIsaac — APPROVED** [review]

**Pull request merged**